projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+                        projectAdmin.push(key);
+                    }
+                }
+
+                // store session information in browser's localstorage
+                // TODO: jwt will be removed, when we have a better cookie solution (DSP-261)
+                session = {
+                    id: this.setTimestamp(),
+                    user: {
+                        name: response.body.user.username,
+                        jwt: jwt,
+                        lang: response.body.user.lang,
+                        sysAdmin: sysAdmin,
+                        projectAdmin: projectAdmin
+                    }
+                };
+
+                // update localStorage
+                localStorage.setItem('session', JSON.stringify(session));
+            },
+            (error: ApiResponseError) => {
+                localStorage.removeItem('session');
+                console.error(error);
+            }
+        );
+    }
+
+    /**
+     * Validate intern session and check knora api credentials if necessary.
+     * If a json web token exists, it doesn't mean that the knora api credentials are still valid.
+     *
+     * @returns boolean
+     */
+    isSessionValid(): boolean {
+        // mix of checks with session.validation and this.authenticate
+        const session = JSON.parse(localStorage.getItem('session'));
+
+        const tsNow: number = this.setTimestamp();
+
+        if (session) {
+
+            this.updateDspApiConnection(session.user.jwt);
+
+            // check if the session is still valid:
+            if (session.id + this.MAX_SESSION_TIME <= tsNow) {
+                // the internal (dsp-ui) session has expired
+                // check if the api credentials are still valid
+
+                this.dspApiConnection.v2.auth.checkCredentials().subscribe(
+                    (response: ApiResponseData<CredentialsResponse>) => {
+                        // the knora api credentials are still valid
+
+                        // update the session.id
+                        session.id = tsNow;
+
+                        localStorage.setItem('session', JSON.stringify(session));
+
+                        return true;
+                    },
+                    (error: ApiResponseError) => {
+                        // a user is not authenticated anymore!
+                        this.destroySession();
+                        return false;
+                    }
+                );
+
+            } else {
+                // the internal (dsp-ui) session is still valid
+                return true;
+            }
+        } else {
+            // no session found; update knora api connection with empty jwt
+            this.updateDspApiConnection();
+            return false;
+        }
+    }
+
+    /**
+     * update the session storage
+     * @param jwt
+     * @param username
+     *
+     * @returns boolean
+     */
+    updateSession(jwt: string, username: string) {
```

Comment:
This method is not used anywhere. Can it be removed?

Reply:
If a system admin removes himself as system admin we have to update the user data in the local storage. The same applies to project admin permissions. A user creates a project and becomes project admin himself. In this case the local storage has to be updated so that the GUI knows what it is allowed to display in the project area and what not.

Reply:
Will be resolved in the app [DSP-449](https://dasch.myjetbrains.com/youtrack/issue/DSP-449)

Reply:
Resolved in [6c9bc68](https://github.com/dasch-swiss/knora-ui-ng-lib/pull/84/commits/6c9bc68bd1e0eb715973d7ee48df8675202883c3)